File: flutter/lib/src/log4agent_endpoint.dart
class Log4AgentEndpoint {
  static const defaultAndroidEmulator = 'http://10.0.2.2:3100/logs';
  static const defaultIosSimulator = 'http://127.0.0.1:3100/logs';

  static String fromHost(
    String host, {
    int port = 3100,
    String path = '/logs',
    String scheme = 'http',
  }) {
    final normalizedPath = path.startsWith('/') ? path : '/$path';
    return '$scheme://$host:$port$normalizedPath';
  }
}

File: android/log4agent/src/main/java/dev/log4agent/Log4AgentEndpoint.kt
package dev.log4agent

object Log4AgentEndpoint {
    fun default(): String = fromHost("10.0.2.2")

    fun fromHost(host: String, port: Int = 3100, path: String = "/logs", scheme: String = "http"): String {
        val normalizedPath = if (path.startsWith("/")) path else "/$path"
        return "$scheme://$host:$port$normalizedPath"
    }
}

